src/render/direct3d/SDL_render_d3d.c

```diff
@@ -1552,13 +1552,32 @@ static int D3D_Reset(SDL_Renderer *renderer)
 static int D3D_SetVSync(SDL_Renderer *renderer, const int vsync)
 {
     D3D_RenderData *data = (D3D_RenderData *)renderer->driverdata;
-    if (vsync) {
-        data->pparams.PresentationInterval = D3DPRESENT_INTERVAL_ONE;
-        renderer->info.flags |= SDL_RENDERER_PRESENTVSYNC;
-    } else {
-        data->pparams.PresentationInterval = D3DPRESENT_INTERVAL_IMMEDIATE;
-        renderer->info.flags &= ~SDL_RENDERER_PRESENTVSYNC;
+    DWORD PresentationInterval;
+    switch (vsync) {
+    case 0:
+        PresentationInterval = D3DPRESENT_INTERVAL_IMMEDIATE;
+    case 1:
+        PresentationInterval = D3DPRESENT_INTERVAL_ONE;
+    case 2:
+        PresentationInterval = D3DPRESENT_INTERVAL_TWO;
+    case 3:
+        PresentationInterval = D3DPRESENT_INTERVAL_THREE;
+    case 4:
+        PresentationInterval = D3DPRESENT_INTERVAL_FOUR;
+    default:
+        PresentationInterval = 0;
```

Comment:
This works because the `!(caps.PresentationIntervals & PresentationInterval)` test below fails, but would it be clearer to just reutrn SDL_Unsupported() here?

Reply:
Yep!